src/rules/at-import-no-partial-leading-underscore/index.js

```diff
@@ -0,0 +1,44 @@
+import { utils } from "stylelint"
+
+export const ruleName = "at-import-no-partial-leading-underscore"
+
+export const messages = utils.ruleMessages(ruleName, {
+  expected: "Unexpected leading underscore in imported partial name",
+})
+
+export default function (actual) {
+  return (root, result) => {
+    const validOptions = utils.validateOptions(result, ruleName, { actual })
+    if (!validOptions) { return }
+
+    function checkPathForUnderscore(path, decl) {
+      // Stripping trailing quotes and whitespaces, if any
+      const pathStripped = path.replace(/^\s*?("|')\s*?/, "").replace(/\s*?("|')\s*?$/, "")
+
+      // Skipping importing CSS: url(), ".css", URI with a protocol, media
+      if (pathStripped.slice(0, 4) === "url(" ||
+        pathStripped.slice(-4) === ".css" ||
+        pathStripped.search("//") !== -1 ||
+        pathStripped.search(/(?:\s|[,)"'])\w+$/) !== -1
+      ) {
+        return
+      }
+      // Searching a _ at the start of filename
+      if (pathStripped.search(/(?:^|\/|\\)_[^/]+$/) === -1) { return }
```

Comment:
Same here, there seems to be a speed boost if you do this before CSS import checks:

```
no leading _ search late x 1,170,845 ops/sec ±3.89% (77 runs sampled)
no leading _ search early x 13,380,512 ops/sec ±2.25% (79 runs sampled)
Fastest is no leading _ search early
```